site/src/util/mdx-components.tsx

```diff
@@ -180,23 +181,45 @@ export const mdxComponents: Record<string, ComponentType> = {
     );
   },
   h4: (props: TypographyProps) => {
+    const anchor = slugify(stringifyChildren(props.children), {
+      lower: true,
+    });
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { headingRef } = usePageHeading({ anchor });
+
     return (
       <Heading
+        ref={headingRef}
         mt={HEADING_MARGIN_TOP.H5}
         mb={HEADING_MARGIN_BOTTOM}
         variant="bpHeading5"
         {...props}
-      />
+      >
+        {props.children}
+        <HeadingAnchor anchor={anchor} depth={4} />
+      </Heading>
     );
   },
   h5: (props: TypographyProps) => {
+    const anchor = slugify(stringifyChildren(props.children), {
+      lower: true,
+    });
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { headingRef } = usePageHeading({ anchor });
+
     return (
       <Heading
+        ref={headingRef}
         mt={HEADING_MARGIN_TOP.H5}
         mb={HEADING_MARGIN_BOTTOM}
         variant="bpHeading5"
         {...props}
-      />
+      >
+        {props.children}
+        <HeadingAnchor anchor={anchor} depth={5} />
+      </Heading>
     );
```

Comment:
Just want to flag that giving h3 and h4 tags anchors breaks the current implementation of the sidebar in a way that it no longer displays which section you're currently in:

https://user-images.githubusercontent.com/42802102/174753716-023c8620-f4e3-4952-8078-3ab967044b8f.mp4


I think this is because it relies on reading the anchor to determine which section the user is currently in, and it is not able to currently infer that from h4 and h5 headings. It's therefore perhap worth considering not adding these anchor tags for now, and creating a follow-up task where these are added and integrated into the sidebar.

Reply:
Just noticed you allude to this in the description, and appreciate the reason you've introduced the anchor tags is so that the methods can link to each other. Just want to make sure you're aware that this breaks the position indicator anywhere an h4 tag is currently being used.

Reply:
I've removed these in https://github.com/blockprotocol/blockprotocol/pull/387/commits/3e8b7528d05f717172be277966b30038d4694135 – I don't think the linking is important enough to figure out the nav right now

Reply:
Makes sense, this way the sidebar won't seem as buggy until we do have a better implementation of it. Thanks for addressing 👍